azure-pipelines/templates/steps/vsts-release.yml

```diff
@@ -0,0 +1,80 @@
+# File: azure-pipelines\templates\steps\nightly-snapshot-releases\vsts-release.yml
+# Description: Template to assemble and publish Snapshot/Latest versions of ${{ parameters.projectName }}
+# Variable: 'ENV_VSTS_MVN_ANDROIDADACCOUNTS_USERNAME' was defined in the Variables tab
+# Variable: 'mvnAccessToken' was defined in the Variables tab
+# Variable: 'folderName' refers to the project folder
+# Variable: 'gradleTasks' list all the gradle tasks to be used.
+# Variable: 'accessTokenName'  secret used to access the private azure devops maven feed.
+# Variable: 'runUnitTest' boolean: refers to whether the pipeline will run unit tests or not.
+# https://docs.microsoft.com/en-us/azure/devops/organizations/accounts/use-personal-access-tokens-to-authenticate
+# Variable: 'versionType' type of version (Latest, Snapshot)
+
+parameters:
+- name: versionType
+  default: ''
+  values:
+  - Snapshot
+  - Latest
+- name: projectName
+  default: none
+- name: gradleTasks
+  default: none
+- name: accessTokenName
+  default: none
+- name: runUnitTest
+  default: false
+
+jobs:
+- job: Phase_1
+  displayName: Initiating Release Process
+  timeoutInMinutes: 90
+  cancelTimeoutInMinutes: 1
+  condition: and(ne(variables['parameters.versionType'], ''), ne(variables['parameters.gradleTasks'], ''))
+  pool:
+    name: Hosted Windows 2019 with VS2019
+  steps:
+  - checkout: self
+    clean: true
+    submodules: recursive
+    persistCredentials: True
+  - task: CmdLine@1
+    displayName: Set MVN Access Token in Environment
+    inputs:
+      filename: echo
+      arguments: '##vso[task.setvariable variable=${{ parameters.accessTokenName }}]$(mvnAccessToken)'
+  - task: Gradle@2
+    displayName: Run unit tests
+    condition: eq(variables['parameters.runUnitTests'], 'true')
+    inputs:
+      tasks: testDebugUnitTest
+  - task: Gradle@2
+    displayName: ${{ parameters.versionType }} version update
```

Comment:
I was referring to make this task conditional

Reply:
because we have the following types of VSTS releases:

- snapshot
- latest
- (none of the above i.e. a regular VSTS releases where we don't perform a version update task)

I feel like this template can be powerful enough to support that